### ISSUE TITLE
Vars request matchers cannot match map variables

### ISSUE THREAD

**Issue by mao-shonen** [caddyserver/caddy#5034]
## Question description
According to the documentation,
I can filter the map variables with vars Request Matchers,
but in fact Request Matchers seems to ignore it?

## Minimum example
```nginx
:80 {
	map {host} {num} {
		"example.com"	1
		default		0
	}

	@match_endpoint vars {num} 1
	handle @match_endpoint {
		respond "test endpoint"
	}

	handle {
		respond "default endpoint num={num}"
	}
}
```

## Test results
```bash
# httpie
http :8080 Host:example.com
```
Expected to get `test endpoint` but
```http
HTTP/1.1 200 OK
Content-Length: 22
Date: Sun, 11 Sep 2022 15:23:34 GMT
Server: Caddy

default endpoint num=1
```

## Context
caddy version: v2.5.2 h1:eCJdLyEyAGzuQTa5Mh3gETnYWDClo1LjtQm2q9RNZrs=

**Comment by mholt** [comment]
If you change `@match_endpoint vars {num} 1` to `@match_endpoint vars num 1` does it work?

It should work the way you're doing it though. Try the latest on master, I think I fixed it a while ago.

**Comment by mao-shonen** [comment]
> If you change `@match_endpoint vars {num} 1` to `@match_endpoint vars num 1` does it work?
> 
> It should work the way you're doing it though. Try the latest on master, I think I fixed it a while ago.

Great! It working!
Is this trick written in the documentation? Am I missing out on reading it?

**Comment by mholt** [comment]
Your first attempt is supposed to work. I just asked you to change it to verify it's the bug I thought it was. It will be fixed in 2.6.
